Source/CesiumRuntime/Private/Cesium3DTileset.cpp

```diff
@@ -587,29 +592,43 @@ void ACesium3DTileset::LoadTileset() {
   this->GetComponents<UCesiumRasterOverlay>(rasterOverlays);
 
   if (pTileset) {
-    if (this->Material == _lastMaterial) {
-      // The material hasn't been changed, check for changes in the url or Ion
-      // asset ID / access token
-      if (this->Url.Len() > 0) {
-        if (pTileset->getUrl() &&
-            TCHAR_TO_UTF8(*this->Url) == pTileset->getUrl()) {
-          // Already using this URL.
-          return;
-        }
-      } else {
-        if (pTileset->getIonAssetID() && pTileset->getIonAccessToken() &&
-            this->IonAssetID == pTileset->getIonAssetID() &&
-            TCHAR_TO_UTF8(*this->IonAccessToken) ==
-                pTileset->getIonAccessToken()) {
-          // Already using this asset ID and access token.
-          return;
-        }
-      }
-    } else {
-      _lastMaterial = this->Material;
+    // The tileset already exists, check if any properties were changed. If
+    // none were changed we can ignore this LoadTileset call.
+
+    // Unlike the set of properties checked in Tick, these properties require
+    // the tileset to be destroyed and recreated when changed.
+
+    // check if the tileset source selection itself changed
+    bool tilesetSourceChanged = this->TilesetSource != this->_lastTilesetSource;
+    this->_lastTilesetSource = this->TilesetSource;
+
+    // check if the current tileset changed
+    switch (this->TilesetSource) {
+    case ETilesetSource::FromUrl:
+      tilesetSourceChanged =
+          pTileset->getUrl() ? TCHAR_TO_UTF8(*this->Url) != pTileset->getUrl()
+                             : this->Url.Len() > 0;
```

Comment:
The way this is written, switching from `FromCesiumIon` to `FromUrl` won't make the Cesium ion tileset immediately disappear if the Url is blank. Is that intended? It seems like it might be a little confusing.